internal/provider/job_resource.go

```diff
@@ -0,0 +1,321 @@
+	return bytes.NewReader(json_raw), nil
+}
+
+// Configure adds the provider configured client to the data source.
+func (d *JobResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*AAPClient)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *AAPClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	d.client = client
+}
+
+func (r JobResource) CreateJob(ctx context.Context, data *jobResourceModel) error {
+
+	// Create new Job from job template
+	req_data, err := data.CreateRequestBody(ctx)
+	if err != nil {
+		return err
+	}
+
+	var http_code int
+	var body []byte
+	post_url := "/api/v2/job_templates/" + data.TemplateId.String() + "/launch/"
+	if req_data != nil {
+		tflog.Info(ctx, fmt.Sprintf("HTTP POST [%s] => %s", post_url, string(body)))
+		http_code, body, err = r.client.doRequestWithBody("POST", post_url, req_data)
+	} else {
+		tflog.Info(ctx, fmt.Sprintf("HTTP POST [%s] => %s", post_url, string(body)))
+		http_code, body, err = r.client.doRequest("POST", post_url)
+	}
+
+	if err != nil {
+		return err
+	}
+	if http_code != http.StatusCreated {
+		return fmt.Errorf("The server returned status code %d while attempting to create Job", http_code)
+	}
+	err = data.ParseHttpResponse(body)
+	if err != nil {
+		return fmt.Errorf("Failed to parse HTTP JSON Response: " + err.Error())
+	}
+	return nil
+}
+
+func (r JobResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data jobResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if !data.URL.IsNull() && !data.URL.IsUnknown() {
+		// Read existing Job
+		http_code, body, err := r.client.doRequest("GET", data.URL.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Unexpected Resource Read error",
+				err.Error(),
+			)
+			return
+		}
+
+		if http_code != http.StatusOK {
+			resp.Diagnostics.AddError(
+				"Unexpected Resource Read error",
+				fmt.Sprintf("The server returned status code %d while attempting to Get from URL %s", http_code, data.URL.ValueString()),
+			)
+			return
+		}
+
+		tflog.Info(ctx, fmt.Sprintf("HTTP GET [%s] => %s", data.URL.ValueString(), string(body)))
+		err = data.ParseHttpResponse(body)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Unexpected Resource Read error",
+				"Failed to parse HTTP JSON Response: "+err.Error(),
+			)
+			return
+		}
+	}
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r JobResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data *jobResourceModel
```

Comment:
I'm guessing this is not doing what you want. The default value for a pointer is `nil`. If you want a pointer to a default `jobResourceModel` then use `data := &jobResourceModel{}`. I would suggest just creating the struct and when you need a pointer to it later, just get it then, similar to how you have done in the `Delete` method.

Reply:
done, thanks